City-of-Helsinki/berth-reservations
# VEN-1271 | Create boat for existing applications

## Description :sparkles:
* Add a command that creates (or updates) boats for existing applications **with leases**
  * If the application has a lease, it assigns the boat to the lease
* Add validation to the boat fields on `BaseApplication` 
* Extract boat creation to separate util

## Issues :bug:
### Closes :no_good_woman:
**[VEN-1271](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1271):** Create boats in database for previously handled applications

## Testing :alembic:
### Automated tests :gear:️
```shell
$ pytest applications/management/commands/create_boat_for_existing_applications.py
```

### Manual testing :construction_worker_man:
1. Create an application
2. Create a lease for that application
3. Move the application to `OFFERED` and the lease to `PAID`
4. Run the command `python manage.py create_boat_for_existing_applications`
5. Check the boats for the customer who submitted the application, there should be a new boat with the fields from the application
